src/lib.rs

```diff
@@ -276,7 +283,17 @@ impl OffersMessageHandler for OfferHandler {
                 log::error!("Invoice request received, payment not yet supported.");
                 None
             }
-            OffersMessage::Invoice(_invoice) => None,
+            OffersMessage::Invoice(invoice) => {
+                let secp_ctx = &Secp256k1::new();
+                match invoice.verify(&self.expanded_key, secp_ctx) {
+                    // TODO: Eventually we can use the returned payment id below to check if this
+                    // payment has been sent twice.
+                    Ok(_payment_id) => Some(OffersMessage::Invoice(invoice)),
```

Comment:
Is it safe to leave this as a todo? Seems like this is _the_ key thing that we need to be worried about, people sending us multiple invoices and us double-paying.  Ignore me if this happens in a later on PR

Reply:
Hmm this might be worth some discussion... It seems to me like a PaymentId is only very useful when we have multiple payments in flight and need help distinguishing them. You're definitely right, we need to make sure we aren't getting double invoice requests. But for now in the lndk-cli pay-offer scenario, where there's only going to be one offer/invoice request in the OfferHandler at a time anyway, one option would be just to reject the invoice if we already updated the OfferState to InvoiceReceived, InvoicePaymentDispatched, or InvoicePaid. (I would add this to the next PR since that's where we add the invoice to a queue.) Thoughts?

Also, as you suggested, I wrote up some more context in the comments

Reply:
Gotcha. Can we update that todo to explain why we can leave this (for now) and still be safe?